src/lib/tcti_ldr.c

```diff
@@ -168,49 +168,40 @@ tcti_conf tcti_get_config(void) {
     };
 
     /* no tcti config supplied, get it from env */
-    char *optstr = getenv (TPM2_PKCS11_TCTI);
-    if (!optstr) {
+    char *env_conf_string = getenv(TPM2_PKCS11_TCTI);
+    if (!env_conf_string) {
         /* nothing user supplied, use default */
         return conf;
     }
 
-    char *split = strchr(optstr, ':');
+    char *conf_string = malloc(strlen(env_conf_string) + 1);
+    strcpy(conf_string, env_conf_string);
```

Comment:
use `strdup()` versus malloc and strcpy

Reply:
You also need to free this. Below you try and get to a single return point, so you'll need a free before the return.

Reply:
1. Agree on using `strdup()` and checking whether memory was successfully allocated.
2. Here is where I'm a little confused and might lack some basic understanding. As noted in my previous comment, it was my intention to free the memory, but calling `free(conf_string)` before returning made all tests fail. My reasoning was that in my implementation, since both fields of `tcti_conf` point to a location in `conf_string`, freeing `conf_string` results in `tcti_conf` no longer having access to its data. This seemed to be confirmed by my tests. Surely I'm misunderstanding something, but I don't quite know what.